File: 卡码/代码随想录/1/09-动态规划/Python/02-背包问题/02-完全背包/LeetCode_0279.py
#279. 完全平方数
from typing import *

class Solution:
  def numSquares(self, n: int) -> int:
    dp:List[int]=[float("inf")-1]*(n+1)
    dp[0]=0
    for i in range(1,floor(sqrt(n))+1):
      for j in range(i*i,n+1):
        dp[j]=min(dp[j],dp[j-i*i]+1)
    return dp[n]

File: 卡码/代码随想录/1/09-动态规划/Js/02-背包问题/02-完全背包/LeetCode_0279.js
//279. 完全平方数
const numSquares = function(n) {
	const dp=new Array(n+1).fill(Number.MAX_VALUE-1);
	dp[0]=0;
	for(let i=1;i*i<=n;i++){
		for(let j=i*i;j<=n;j++){
			dp[j]=Math.min(dp[j],dp[j-i*i]+1);
		}
	}
	return dp[n];
};